### PR TITLE
Latest Freight Work To Be Merged

### PR DESCRIPTION
**Pull request by haitamlaarabi**
This change is [Reviewable](https://reviewable.io/reviews/LBNL-UCB-STI/beam/3761)

### CODE REVIEW

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 1]
```diff
@@ -377,12 +376,18 @@ trait BeamHelper extends LazyLogging with BeamValidationHelper {
...
 
   def vehicleEnergy(beamConfig: BeamConfig, vehicleTypes: Map[Id[BeamVehicleType], BeamVehicleType]): VehicleEnergy = {
-    val baseFilePath = Paths.get(beamConfig.beam.agentsim.agents.vehicles.vehicleTypesFilePath).getParent
+    var vehiclePaths = IndexedSeq(
+      Paths.get(beamConfig.beam.agentsim.agents.vehicles.vehicleTypesFilePath).getParent.toString
+    )
+    beamConfig.beam.agentsim.agents.freight.vehicleTypesFilePath
+      .map(freightVehicleTypesFilePath => Paths.get(freightVehicleTypesFilePath).getParent.toString)
+      .foreach(freightVehiclePath => vehiclePaths = vehiclePaths :+ freightVehiclePath)
+
```
Comment (dimaopen) [inline src/main/scala/beam/sim/BeamHelper.scala]:
```suggestion
    val vehiclePaths =
      Paths.get(beamConfig.beam.agentsim.agents.vehicles.vehicleTypesFilePath).getParent.toString +:
      beamConfig.beam.agentsim.agents.freight.vehicleTypesFilePath
        .map(freightVehicleTypesFilePath => Paths.get(freightVehicleTypesFilePath).getParent.toString)
        .toIndexedSeq
```
I personally find hard to understand when an external collection gets changed in a loop. Hence, I would suggest this change.

File: src/main/scala/beam/agentsim/agents/freight/input/GenericFreightReader.scala [hunk 2]
```diff
@@ -183,15 +183,19 @@ class GenericFreightReader(
       case Some(filePath) => vehicleTypes ++ readBeamVehicleTypeFile(filePath)
       case None           => vehicleTypes
     }
-
-    val existingTours: Set[Id[FreightTour]] = allTours.keySet.intersect(allPlans.map(_._2.tourId).toSet)
+    // ****
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/freight/input/GenericFreightReader.scala:186]: I suggest adding a comment or removing it.

**Review by dimaopen — APPROVED** [review]

File: src/main/scala/beam/agentsim/agents/freight/input/GenericFreightReader.scala (hunk 2 above)
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/freight/input/GenericFreightReader.scala:186]: I prefer your suggestion. I committed your suggestion.

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Pull request merged**